### ISSUE TITLE
Bigint support

### ISSUE THREAD

**Issue by snowinmars** [convert-units/convert-units#204]
It's a discussion: how to support `bigint` as well as `number`?

Current state:
- `convert(90071992547409910000).from('km').to('m')` won't work *properly* due to js max safe integer restriction,
- `convert(90071992547409910000n).from('km').to('m')` returns zero.

I see two approaches:
- create a second builder, maybe as separate npm package: `convertBigint(value: bigint): BigintConverter<TMeasures, TSystems, TUnits>`,
- use bigint as default type for values - but I don't really like it.

I can create a draft pull request after the discussion, if this feature is required.

There's an extra meme: `jest` doesn't have full `bigint` support as well)

**Comment by Taar** [comment]
While I see the benefits in supporting something like this, unfortunately I don't think it's worth adding to the project at this time. See my reasoning/explanation below.

This would definitely need to be a separate package as it's likely that a third party package will be required to support `bigint`. I say this because `bigint` is an integer type, which means it [does not support decimals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt). To get decimal support, the project would have to roll it's own type or enforce the use of [JSBI](https://github.com/GoogleChromeLabs/jsbi). Further discussion on the topic can be found [here](https://stackoverflow.com/questions/54409854/how-to-divide-two-native-javascript-bigints-and-get-a-decimal-result). The latter would be the best choice since rolling a custom decimal type from scratch *could* be difficult to maintain (imo). This would also mean that users of the package would have to be willing to adopt `JSBI` in their projects or our custom decimal object. That might be a deal breaker for some folks.

There's also no guarantee that `JSBI` will continue to support decimal numbers as it's not part of the [`bigint` spec](https://tc39.es/proposal-bigint/#sec-numeric-types-bigint-divide).

Additional discussion about a [`BigDecimal` type](https://stackoverflow.com/questions/16742578/bigdecimal-in-javascript/66939244#66939244) in javascript. 

In short, unless there is a large demand for supporting numbers larger than the integer max, I think this sort of feature should be placed on the back burner for now.

**Comment by serdaryesilmurat** [comment]
We have a similar issue in our project for time conversions as well. For instance, 1 second becomes 999999999,9999999 ns after the conversion. If this library does not aim for scientific usage or being as accurate as possible, I guess you may skip this issue :pensive:

**Comment by carlocorradini** [comment]
@Taar 
It would be super useful, but I understand the effort needed to add this feature to the library

**Comment by Taar** [comment]
Going to close this issue as there is no plans to support `bigint`.  Maybe in the future this feature can be revisited.
